### PR TITLE
Add ONNX Export Support to rsqrt

### PR DESCRIPTION
**Pull request by lara-hdr**

### CODE REVIEW

File: torch/onnx/symbolic_opset9.py [hunk 1]
```diff
@@ -163,6 +163,10 @@ def sqrt(g, self):
     return g.op("Sqrt", self)
 
 
+def rsqrt(g, self):
+    return div(g, g.op("Constant", value_t=torch.tensor(1.)), sqrt(g, self))
```
Comment (spandantiwari) [inline torch/onnx/symbolic_opset9.py]: If type of self is double, is there a possibility of type mismatch between the two arguments of the `div` op given that `Constant` is always of type float?

**Review by spandantiwari — CHANGES_REQUESTED** [review]

File: torch/onnx/symbolic_opset9.py (hunk 1 above)
Comment (lara-hdr) [reply torch/onnx/symbolic_opset9.py]: I am creating a constant of the correct dtype now. Thanks!

**Comment by lara-hdr** [comment]
CI failing because the test has a case of rsqrt with a double input, which is supported in ORT v0.5.0.
Waiting on ORT version bump in https://github.com/pytorch/pytorch/pull/24414 to fix the CI failure.

File: torch/onnx/symbolic_opset9.py [hunk 2]
```diff
@@ -163,6 +163,13 @@ def sqrt(g, self):
     return g.op("Sqrt", self)
 
 
+def rsqrt(g, self):
+    dtype = self.type().scalarType()
+    dtype = sym_help.scalar_type_to_onnx.index(sym_help.cast_pytorch_to_onnx[dtype])
```
[the remark this answers is not in the document]
Comment (BowenBao) [reply torch/onnx/symbolic_opset9.py]: Actually for this particular use case you can use the `_if_scalar_type_as` method in symbolic_helper.py

**Comment by houseroad** [comment]
Can we rebase to the current master? Some unrelated tests are failing.

**Comment by houseroad** [comment]
```
zip(outputs, ort_outs)]
Sep 06 20:13:27 E   AssertionError: 
Sep 06 20:13:27 E   Not equal to tolerance rtol=0.001, atol=1e-07
Sep 06 20:13:27 E   
Sep 06 20:13:27 E   Mismatch: 0.529%
Sep 06 20:13:27 E   Max absolute difference: 1.7881393e-07
Sep 06 20:13:27 E   Max relative difference: nan
Sep 06 20:13:27 E    x: array([[[ 7.595517e-02,  2.509025e-01,  1.384402e-01],
Sep 06 20:13:27 E           [ 1.481673e-01,  4.209695e-01,  1.445810e-01],
Sep 06 20:13:27 E           [ 3.442400e-01,  1.058863e-01,  4.277925e-02],...
Sep 06 20:13:27 E    y: array([[[ 7.595515e-02,  2.509025e-01,  1.384403e-01],
Sep 06 20:13:27 E           [ 1.481673e-01,  4.209695e-01,  1.445810e-01],
Sep 06 20:13:27 E           [ 3.442400e-01,  1.058862e-01,  4.277921e-02],...
```

**Comment by lara-hdr** [comment]
> ```
> zip(outputs, ort_outs)]
> Sep 06 20:13:27 E   AssertionError: 
> Sep 06 20:13:27 E   Not equal to tolerance rtol=0.001, atol=1e-07
> Sep 06 20:13:27 E   
> Sep 06 20:13:27 E   Mismatch: 0.529%
> Sep 06 20:13:27 E   Max absolute difference: 1.7881393e-07
> Sep 06 20:13:27 E   Max relative difference: nan
> Sep 06 20:13:27 E    x: array([[[ 7.595517e-02,  2.509025e-01,  1.384402e-01],
> Sep 06 20:13:27 E           [ 1.481673e-01,  4.209695e-01,  1.445810e-01],
> Sep 06 20:13:27 E           [ 3.442400e-01,  1.058863e-01,  4.277925e-02],...
> Sep 06 20:13:27 E    y: array([[[ 7.595515e-02,  2.509025e-01,  1.384403e-01],
> Sep 06 20:13:27 E           [ 1.481673e-01,  4.209695e-01,  1.445810e-01],
> Sep 06 20:13:27 E           [ 3.442400e-01,  1.058862e-01,  4.277921e-02],...
> ```

Where do you see this failure?
I added a test with zeros in the input as well to double check that the error is not happening because of a division by zero, but it is passing.

**Comment by houseroad** [comment]
https://circleci.com/api/v1.1/project/github/pytorch/pytorch/2701962/output/105/0?file=true

**Comment by lara-hdr** [comment]
> https://circleci.com/api/v1.1/project/github/pytorch/pytorch/2701962/output/105/0?file=true

yes, the failure in TestONNXRuntime.test_elman_relu_trilayer_forward_with_initial_state_with_variable_length_sequences_with_dropout is unrelated and should be fixed by #25205

**Review by houseroad — COMMENTED** [review]
Please check whether both output are nan. If so we may want to enable the close function to allow nan.

**Comment by houseroad** [comment]
Or we can change the input range to make sure no rsqrt is generated.

**Comment by lara-hdr** [comment]
> Please check whether both output are nan. If so we may want to enable the close function to allow nan.

@houseroad I am not sure what you are referring to, all the added tests for rsqrt are succeeding, or am I missing something?

**Comment by houseroad** [comment]
nvm, it's already merged :-)

@pytorchbot retest this please

**Comment by houseroad** [comment]
Ah, thanks... it should be rebase... :-)

**Review by houseroad — APPROVED** [review]
LGTM

**Pull request closed without merging**